File: client/bcephal-client-model/Sheets/Cell.cs
using Bcephal.Models.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Sheets
{
    public class Cell
    {

        public int Row { get; set; }
        public int Column { get; set; }

        public int Sheet { get; set; }

        private string name;
        public string Name 
        {
            get
            {
                if (string.IsNullOrWhiteSpace(name))
                {
                    name = RangeUtil.GetCellName(Row, Column);
                }
                return name;
            }
            set { name = value; }
        }

        public Cell() { }

        public Cell(int row, int column, int sheet, string name)
        {
            this.Row = row;
            this.Column = column;
            this.Sheet = sheet;
            this.Name = name;
        }

        public override string ToString()
        {
            return this.Name;
        }

    }
}

File: server/bcephal-sheet/src/main/java/com/moriset/bcephal/sheet/domain/Cell.java
/**
 * 
 */
package com.moriset.bcephal.sheet.domain;

import lombok.Data;

/**
 * @author Joseph Wambo
 *
 */
@Data
public class Cell {

	private int row;

	private int column;

	private int sheet;

	private String name;
//    public string Name 
//    {
//        get
//        {
//            if (string.IsNullOrWhiteSpace(name))
//            {
//                name = RangeUtil.GetCellName(Row, Column);
//            }
//            return name;
//        }
//        set { name = value; }
//    }

	public Cell() {
	}

	public Cell(int row, int column, int sheet, String name) {
		this.row = row;
		this.column = column;
		this.sheet = sheet;
		this.name = name;
	}

	@Override
	public String toString() {
		return this.name;
	}

}
